html/admin/api/server/migrateAsset.php

```diff
@@ -0,0 +1,101 @@
+<?php
+require_once __DIR__ . '/../apiHeadSecure.php';
+
+if (!$AUTH->serverPermissionCheck("INSTANCES:FULL_PERMISSIONS_IN_INSTANCE") or !isset($_POST['assets_id']) or !isset($_POST['instances_id'])) die("404");
+
+//Get asset
+$DBLIB->where("assets_id", $_POST['assets_id']);
+$DBLIB->where("assets_deleted", 0);
+$DBLIB->where("assets_archived IS NULL");
+$asset = $DBLIB->getOne("assets");
+if (!$asset) finish(false, ["code"=>"ASSET_NOT_FOUND", "message"=>"Asset not found"]);
+
+//Get assetType
+$DBLIB->where("assetTypes_id", $asset['assetTypes_id']);
+$assetType = $DBLIB->getOne("assetTypes", ["assetTypes.*"]);
+if (!$assetType) finish(false, ["code"=>"ASSET_TYPE_NOT_FOUND", "message"=>"Asset type not found"]);
+
+//Check if asset is already in instance
+if ($assetType['instances_id'] == $_POST['instances_id']) finish(false, ["code"=>"ASSET_ALREADY_IN_INSTANCE", "message"=>"Asset already in instance"]);
```

Comment:
This just says that the asset type is in the instance right - rather than the asset itself? This means you can't move an asset across if there is already an asset of that type in the instance